gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -444,41 +444,42 @@ protected static MetricNameRegexFilter getMetricsFilterForDagManager() {
     private final MetricContext metricContext;
     private final Optional<EventSubmitter> eventSubmitter;
     private final Optional<Timer> jobStatusPolledTimer;
-    private final int defaultQuota;
-    private final Map<String, Integer> perUserQuota;
     private final AtomicLong orchestrationDelay = new AtomicLong(0);
     private static final Map<String, FlowState> flowGauges = Maps.newConcurrentMap();
     private final ContextAwareMeter allSuccessfulMeter;
     private final ContextAwareMeter allFailedMeter;
     private static final Map<String, ContextAwareMeter> groupSuccessfulMeters = Maps.newConcurrentMap();
     private static final Map<String, ContextAwareMeter> groupFailureMeters = Maps.newConcurrentMap();
-
+    private final GobblinServiceQuotaManager quotaManager;
     private final JobStatusRetriever jobStatusRetriever;
     private final DagStateStore dagStateStore;
     private final DagStateStore failedDagStateStore;
     private final BlockingQueue<Dag<JobExecutionPlan>> queue;
     private final BlockingQueue<String> cancelQueue;
     private final BlockingQueue<String> resumeQueue;
     private final Long defaultJobStartSlaTimeMillis;
+    private final Set<String> runningDags;
     /**
      * Constructor.
      */
     DagManagerThread(JobStatusRetriever jobStatusRetriever, DagStateStore dagStateStore, DagStateStore failedDagStateStore,
         BlockingQueue<Dag<JobExecutionPlan>> queue, BlockingQueue<String> cancelQueue, BlockingQueue<String> resumeQueue,
-        boolean instrumentationEnabled, int defaultQuota, Map<String, Integer> perUserQuota, Set<String> failedDagIds,
-        ContextAwareMeter allSuccessfulMeter, ContextAwareMeter allFailedMeter, Long defaultJobStartSla) {
+        boolean instrumentationEnabled, Set<String> failedDagIds, ContextAwareMeter allSuccessfulMeter,
+        ContextAwareMeter allFailedMeter, Long defaultJobStartSla, GobblinServiceQuotaManager quotaManager,
+        Set<String> runningDags) {
       this.jobStatusRetriever = jobStatusRetriever;
       this.dagStateStore = dagStateStore;
       this.failedDagStateStore = failedDagStateStore;
       this.failedDagIds = failedDagIds;
       this.queue = queue;
       this.cancelQueue = cancelQueue;
       this.resumeQueue = resumeQueue;
-      this.defaultQuota = defaultQuota;
-      this.perUserQuota = perUserQuota;
       this.allSuccessfulMeter = allSuccessfulMeter;
       this.allFailedMeter = allFailedMeter;
       this.defaultJobStartSlaTimeMillis = defaultJobStartSla;
+      this.quotaManager = quotaManager;
+      this.runningDags = runningDags;
```

Comment:
these two co-occur, which makes sense.  any reason not to encapsulate the notion of `runningDags` within the `quotaManager`?  you could still have a method there to provide the current count of running DAGs (for monitoring/metrics)

Reply:
there may even be reason for coordination between the two, such that a more sophisticated user quota / rate limiting could take into account the overall system load (potentially to allow some overrun to accommodate temp burstiness).